### PR TITLE
Initial MSan library builder script.

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

File: infra/base-images/msan-builder/compiler_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+from __future__ import print_function
+import os
+import sys
```
Comment (Dor1s) [inline infra/base-images/msan-builder/compiler_wrapper.py]: nit: alpha order: subprocess, then sys

File: infra/base-images/msan-builder/msan_build.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, 'clang'))
+
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, 'clang++'))
+
+  env['CC'] = os.path.join(bin_dir, 'clang')
+  env['CXX'] = os.path.join(bin_dir, 'clang++')
+
+  # Not all build rules respect $CC/$CXX, so make additional symlinks.
```
Comment (Dor1s) [inline infra/base-images/msan-builder/msan_build.py]: would it also make sense to try setting other "popular" env variables like `CPP`, `cc`, `cpp`, `cxx`?

File: infra/base-images/msan-builder/msan_build.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  dpkg_host_architecture = subprocess.check_output(
+      ['dpkg-architecture', '-qDEB_HOST_GNU_TYPE']).strip()
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, dpkg_host_architecture + '-gcc'))
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, dpkg_host_architecture + '-g++'))
+
+  os.symlink(compiler_wrapper_path, os.path.join(bin_dir, 'gcc'))
+  os.symlink(compiler_wrapper_path, os.path.join(bin_dir, 'cc'))
```
Comment (Dor1s) [inline infra/base-images/msan-builder/msan_build.py]: same question here, I've seen `c++` somewhere, though not sure if we should try to guess as much as possible

File: infra/base-images/msan-builder/compiler_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    return compiler_args
+
+  # FORTIFY_SOURCE is not supported by sanitizers.
+  compiler_args.extend([
+      '-U_FORTIFY_SOURCE',
+  ])
+
+  REMOVED_ARGS = [
+      '-Wl,-z,defs',
```
Comment (Dor1s) [inline infra/base-images/msan-builder/compiler_wrapper.py]: Is it possible that argument like this has a white-space character?

File: infra/base-images/msan-builder/compiler_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def GetCompilerArgs(args):
+  compiler_args = args[1:]
+
+  if Is32Bit(args):
+    # 32 bit builds not supported.
+    compiler_args.extend([
+        '-fno-sanitize=memory',
+        '-fno-sanitize-memory-track-origins',
```
Comment (Dor1s) [inline infra/base-images/msan-builder/compiler_wrapper.py]: Just to make sure: these arguments are position independent, right? Other compilations flags cannot overwrite them?

File: infra/base-images/msan-builder/compiler_wrapper.py (hunk 1 above)
Comment (oliverchang) [reply infra/base-images/msan-builder/compiler_wrapper.py]: Done!

File: infra/base-images/msan-builder/compiler_wrapper.py (hunk 5 above)
Comment (oliverchang) [reply infra/base-images/msan-builder/compiler_wrapper.py]: These are added to the very end, so should override any previous compilation flags.

File: infra/base-images/msan-builder/compiler_wrapper.py (hunk 4 above)
Comment (oliverchang) [reply infra/base-images/msan-builder/compiler_wrapper.py]: I don't think spaces work here.

File: infra/base-images/msan-builder/msan_build.py (hunk 2 above)
Comment (oliverchang) [reply infra/base-images/msan-builder/msan_build.py]: Hmm, let's stick with CC/CXX for now and see if we miss any.

File: infra/base-images/msan-builder/msan_build.py (hunk 3 above)
Comment (oliverchang) [reply infra/base-images/msan-builder/msan_build.py]: Sure, added `c++` since I think that's pretty popular.

**Pull request merged**